include/toolchain/common.h

```diff
@@ -187,6 +187,9 @@
  */
 #define Z_DECL_ALIGN(type) __aligned(__alignof(type)) type
 
+/* Check if a pointer is aligned enough for a particular data type. */
+#define IS_PTR_ALIGNED(ptr, type) ((((uintptr_t)ptr) % __alignof(type)) == 0)
```

Comment:
@rlubos it would be possible to replace some of the `UNALIGNED_GET()` code in networking with usage of this macro inside an assert statement.

Reply:
Do you have any examples where could this be applied? I had an impression, that with data inside a packet (which the packed struct is casted on) we should not make any assumptions regarding alignment?

Reply:
No, only in places where you know or suspect that the alignment has been ensured by the allocation itself. If you do not know where the allocation was performed then it wouldn't be useful. But I thought we had some of those in networking, maybe I was wrong.